src/parser/warlock/affliction/modules/azerite/InevitableDemise.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Entities from 'parser/shared/modules/Entities';
+import Events from 'parser/core/Events';
+import calculateBonusAzeriteDamage from 'parser/core/calculateBonusAzeriteDamage';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+import SPELLS from 'common/SPELLS';
+import { calculateAzeriteEffects } from 'common/stats';
+import { formatThousands } from 'common/format';
+
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import ItemDamageDone from 'interface/others/ItemDamageDone';
+
+const DRAIN_LIFE_SP_COEFFICIENT = 0.12; // taken from Simcraft SpellDataDump
+const debug = false;
+
+class InevitableDemise extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  _currentStacks = 0;
+  damagePerStack = 0;
+  damage = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.INEVITABLE_DEMISE.id);
+    if (!this.active) {
+      return;
+    }
+    this.damagePerStack = this.selectedCombatant.traitsBySpellId[SPELLS.INEVITABLE_DEMISE.id].reduce((total, rank) => {
+      const [ damage ] = calculateAzeriteEffects(SPELLS.INEVITABLE_DEMISE.id, rank);
+      return total + damage;
+    }, 0);
+    debug && this.log('Total damage per ID stack', this.damagePerStack);
+
+    this.addEventListener(Entities.changebuffstack.by(SELECTED_PLAYER).spell(SPELLS.INEVITABLE_DEMISE_BUFF), this.onIDChangeBuffStack);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DRAIN_LIFE), this.onDrainLifeDamage);
+  }
+
+  onIDChangeBuffStack(event) {
```

Comment:
might be simpler to use 
```
const buff = this.selectedCombatant.getBuff(SPELLS.INEVITABLE_DEMISE_BUFF.id);
const buffStacks = buff ? buff.stacks || 0;
```
to get the buffstacks on the dmg event instead of tracking all changebuffs.